LECTURE: Development Environments Updated and Added

spulick opened the pull request:
fixes #279 

- incorporated @jstac comments
- included content on using Jupyter notebooks in VS Code

spulick committed:
lecture on devEnvs

added lecture on working with python scripts and workspaces

spulick committed:
Update workspace.md

spulick committed:
implemented recommendations and corrections

spulick committed:
Update workspace.md

AakashGfude committed:
Binder hub url removal #281

jstac commented:
Thanks @orectique .  Could you please look into the failing link check?

spulick commented:
I've verified that the links flagged by the link checker are all functional. 

It might be that the flagged sites are shutting down automated access.

jstac commented:
@mmcky or @HumphreyYang , could you please advise?  I think we just ignore the link checker in this case?

HengchengZhang commented:
Hi @jstac, @mmcky and @HumphreyYang, I checked these links manually and they all should be false positives. My opinion is to add them to the ignore list.

HumphreyYang commented:
> Hi @jstac, @mmcky and @HumphreyYang, I checked these links manually and they all should be false positives. My opinion is to add them to the ignore list.

Many thanks @HengchengZhang and @jstac. I also think we can have it on since it still works on other links. We can manually check the white list when we have major releases as long as the list is not too long. We can consider disabling it in the future if the list of false positives is getting too long.

jstac commented:
Thanks @HengchengZhang and @HumphreyYang .  Let's ignore as suggested.

jstac commented:
@orectique Nice work on this lecture.  Let's meet up and discuss it.  I'll contact you to set up a time.

HengchengZhang committed:
Fix broken links

HengchengZhang commented:
Hi @jstac I added them to the ignore list and it's working now.

spulick commented:
Thanks, @jstac.

spulick committed:
suggestions

spulick committed:
Merge branch 'dev-envs-lecture' of https://github.com/QuantEcon/lecture-python-programming.myst into dev-envs-lecture

spulick committed:
Update workspace.md

spulick committed:
add screenshots

spulick committed:
Merge branch 'dev-envs-lecture' of https://github.com/QuantEcon/lecture-python-programming.myst into dev-envs-lecture

spulick committed:
update lectures

Workspaces - implement suggestions and corrections

getting stared - add content on debugging

mmcky commented:
thanks @orectique -- this looks like it is in-work but let me know if you need a review etc.

spulick committed:
implement updates

- getting_started.md: add content on real time collaboration
- workspace.md: correct and revise content

spulick commented:
@mmcky This needs a review, please. I've added new content to the Getting started lecture and revised the workspaces lecture.

HumphreyYang committed:
Migrate Sympy Lecture from the Intro Series (#284)

* add sympy

* update sympy exercise

* Update lectures/sympy.md

Co-authored-by: Smit Lunagariya <smitlunagariya.mat18@itbhu.ac.in>

* Update lectures/sympy.md

Co-authored-by: Smit Lunagariya <smitlunagariya.mat18@itbhu.ac.in>

* update stats

* remove user expressions

* Update lectures/sympy.md

Co-authored-by: zhc <79777246+HengchengZhang@users.noreply.github.com>

* try build without scaling headers

* draft of application section

* update exercise again

* proofread

* proofread changes

* Update lectures/sympy.md

Co-authored-by: mmcky <mmcky@users.noreply.github.com>

* update two sentences

* integrate feedbacks

---------

Co-authored-by: Smit Lunagariya <smitlunagariya.mat18@itbhu.ac.in>
Co-authored-by: zhc <79777246+HengchengZhang@users.noreply.github.com>
Co-authored-by: mmcky <mmcky@users.noreply.github.com>

mmcky committed:
MAINT: Migrate from EC2 backend to GA backend (#285)

* MAINT: Migrate from EC2 backend to GA backend

* update machine details to github actions

mmcky committed:
FIX: fix requests issue with docutils incompatibility (#287)

* FIX: fix requests issue with docutils incompatibility

* Fix broken links

---------

Co-authored-by: hengchengzhang <hengcheng.zhang@outlook.com>

mmcky committed:
MAINT: re-install requests to see if that fixes the malformed requests file

HumphreyYang committed:
Remove MatplotlibDeprecationWarning in Matplotlib Lecture (#289)

* remove MatplotlibDeprecationWarning

* update false-positive link

mmcky commented:
Sorry @orectique this slipped through my radar. Can you update on the latest main and then ping me for review. Thanks,.

mmcky committed:
Update anaconda env (#290)

spulick committed:
Merge branch 'main' into dev-envs-lecture

spulick commented:
Hi @mmcky,

Thanks for the review. I have merged changes on Main to this branch. There should be no conflicts.

mmcky commented on .vscode/settings.json:
@orectique can you remove this vscode profile and add to `.gitignore`

mmcky commented on .ipynb_checkpoints/Demo-checkpoint.ipynb:
@orectique can you please remove `.ipynb_checkpoints` and add to `.gitignore`

mmcky commented on lectures/getting_started.md:
````diff
@@ -382,6 +382,33 @@ Now we `Shift+Enter` to produce this
+
+If you are using a newer version of Jupyter, you should see a bug icon on the right end of the toolbar.
+
+```{figure} /_static/lecture_specific/getting_started/debug.png
````
@orectique can we improve the quality i.e. dpi (or make the image smaller)?

mmcky commented on lectures/workspace.md:
````diff
@@ -0,0 +1,375 @@
+</div>
+```
+
+# Python Development: Writing Longer Programs
````
@orectique I would vote for just `Writing Longer Programs`

mmcky commented on lectures/workspace.md:
```diff
@@ -0,0 +1,375 @@
+
+While they are efficient and adaptable when working with short pieces of code, Notebooks are not the best choice for longer programs and scripts. 
+
+While Jupyter Notebooks can help execute chunks of code one at a time, scripts allow for long pieces of code to be written and executed in a single go. 
```
```suggestion
Jupyter Notebooks are well suited to interactive computing (i.e. data science workflows) and can help execute chunks of code one at a time. Text files and scripts allow for long pieces of code to be written and executed in a single go. 
```

spulick committed:
update suggestions

mmcky commented:
thanks @orectique I reviewed [95de21e](https://github.com/QuantEcon/lecture-python-programming.myst/pull/283/commits/95de21e3deb7e6708ca8472292106aee242024f4) and see they resolved all the outstanding items.  Thank you.

mmcky committed:
mmcky edits

mmcky commented:
thanks @orectique -- @jstac this is now at review stage.

jstac commented:
Thanks @mmcky , looks good to me.  It's nicely written.

mmcky commented:
thanks @jstac and @orectique for these improvements.
